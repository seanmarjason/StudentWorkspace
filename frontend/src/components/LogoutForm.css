.logoutForm {
  position: fixed;
  top: 10px;
  right: 10px;
}

.logoutForm button {
  padding: 1vh 1vw;
  background-color: rgb(86, 86, 255);
  color: white;
}

.logoutForm button:hover {
  cursor: pointer;
}
