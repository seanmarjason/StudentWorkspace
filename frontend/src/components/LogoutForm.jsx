import axios from 'axios';

// Logout form handler
function handleLogoutSubmit(event, callback) {
  event.preventDefault()
  const url = 'http://localhost:3000/users/logout';

  axios.delete(url)
  .then((response) => {
    if (response.status === 200 && response.data === 'Logout successful!') {
      console.log('Logout successful!');
      callback(false);
    }
  })
  .catch((error) => {
    console.log(error.response.status);
  });
}

const LogoutForm = ({ callback }) =>
  <form onSubmit={(event) => handleLogoutSubmit(event, callback)}>
  <div className="button-container">
    <button type="submit">Logout</button>
  </div>
  </form>  

export { LogoutForm }
