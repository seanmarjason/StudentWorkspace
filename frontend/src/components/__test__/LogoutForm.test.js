import {render } from '@testing-library/react';
import { LogoutForm } from '../LogoutForm';

describe('Logout Form' , () => {
  it('should match snapshot', async () => {
    const view = render(<LogoutForm />)

    expect(view.asFragment()).toMatchSnapshot();
  });
  
})
  