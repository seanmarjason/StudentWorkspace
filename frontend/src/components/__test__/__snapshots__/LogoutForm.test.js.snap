// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout Form should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="logoutForm"
  >
    <form>
      <div
        class="button-container"
      >
        <button
          type="submit"
        >
          Logout
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
